File: client/python/tests/pages/ios_contact_list_page.py
"""Page Object — iOS Contacts app: contact list screen."""

from __future__ import annotations

from maestro_runner.models import ExecutionResult

from tests.pages import BasePage
from tests.pages.ios_edit_contact_page import IOSContactEditPage


class IOSContactListPage(BasePage):
    """The iOS Contacts list screen."""

    APP_ID = "com.apple.MobileAddressBook"

    def launch(self) -> ExecutionResult:
        result = self.client.launch_app(self.APP_ID, stop_app=True)
        self.wait_for_animation()
        return result

    def open_create_contact(self) -> IOSContactEditPage:
        self.client.tap(text="Add")
        self.wait_for_animation()
        return IOSContactEditPage(self.client)

    def assert_contact_visible(self, name: str) -> ExecutionResult:
        return self.client.assert_visible(text=name)

File: client/typescript/tests/pages/IOSContactListPage.ts
/** Page Object — iOS Contacts app: contact list screen. */

import { MaestroClient, ExecutionResult } from "../../src";
import { BasePage } from "./BasePage";
import { IOSEditContactPage } from "./IOSEditContactPage";

export class IOSContactListPage extends BasePage {
  static readonly APP_ID = "com.apple.MobileAddressBook";

  constructor(client: MaestroClient) {
    super(client);
  }

  async launch(): Promise<ExecutionResult> {
    const result = await this.client.launchApp(IOSContactListPage.APP_ID, { stopApp: true });
    await this.waitForAnimation();
    return result;
  }

  async openCreateContact(): Promise<IOSEditContactPage> {
    await this.client.tap({ text: "Add" });
    await this.waitForAnimation();
    return new IOSEditContactPage(this.client);
  }

  async assertContactVisible(name: string): Promise<ExecutionResult> {
    return this.client.assertVisible({ text: name });
  }
}